packages/api/src/controllers/helpers.ts

```diff
@@ -699,6 +699,33 @@ export const triggerCatalystPullStart =
         throw new Error(`failed to trigger catalyst pull`);
       };
 
+export const getCatalystStreamMetadata =
```

Comment:
super-nit: wdyt of:
- Renaming this to `waitCatalystStreamReady` since this function doesn't just make a single request
- Moving the `metadataUrl` construction to this function instead, since it is the one that has the logic on how the metadata works. It could receive the `stream` object instead of the `url` and simplify usage